jsonargparse_tests/test_util.py

```diff
@@ -677,3 +678,50 @@ def parse_args(args=[]):
     with pytest.raises(CaptureParserException) as ctx:
         capture_parser(lambda: None)
     ctx.match("No parse_args call to capture the parser")
+
+
+def test_std_input_path():
+    input_text_to_test = "a text here\n"
+    with patch("sys.stdin", StringIO(input_text_to_test)):
+        path = Path("-", mode="")
+        assert path == "-"
+        assert input_text_to_test == path.get_content("r")
+        assert path.std_io
+
+    with patch("sys.stdin", StringIO(input_text_to_test)):
+        path = Path("-", mode="")
+        with path.open("r") as std_input:
+            assert input_text_to_test == "".join([line for line in std_input])
+
+    with patch("sys.stdin", StringIO(input_text_to_test)):
+        path = Path("-", mode="r")
+        with path.open("r") as std_input:
+            assert input_text_to_test == "".join([line for line in std_input])
+
+    with patch("sys.stdin", StringIO(input_text_to_test)):
+        path = Path("-", mode="fr")
+        with path.open("r") as std_input:
+            assert input_text_to_test == "".join([line for line in std_input])
+
+
+def test_std_output_path():
+    path = Path("-", mode="")
+    assert path == "-"
+    with patch("sys.stdout", StringIO("")):
+        with path.open("w") as std_output:
+            std_output.write("test\n")
+    assert path.std_io
+
+    path = Path("-", mode="w")
+    assert path == "-"
+    with patch("sys.stdout", StringIO("")):
+        with path.open("w") as std_output:
+            std_output.write("test\n")
+    assert path.std_io
+
+    path = Path("-", mode="fw")
+    assert path == "-"
+    with patch("sys.stdout", StringIO("")):
```

Comment:
Save the `StringIO` instance to a variable, and after the write, assert that the content of the `StringIO` is what got printed.